### PR TITLE
Import new ppx scheme from let-def

### PR DESCRIPTION
**Pull request by hhugo**
It should provide better merlin support.

### CODE REVIEW

**Comment by Drup** [comment]
Ah, finally! :) 
I'll review next week.

**Comment by hhugo** [comment]
I've tuned locations a little bit. I'm quite happy with `merlin-type-enclosing` now.

**Comment by Drup** [comment]
Hum, the location in the error messages are not really the right one, the one that were there before seems better.

**Comment by hhugo** [comment]
I'll update the PR later this week.

**Comment by hhugo** [comment]
@Drup, I've updated the location.

**Comment by hhugo** [comment]
@Drup, do you have time for review ?

File: lib/ppx/ppx_js.cppo.ml [hunk 1]
```diff
@@ -23,43 +23,35 @@ let exp_to_string = function
...
 
 let inside_Js = lazy
   (try
      Filename.basename (Filename.chop_extension !Location.input_name) = "js"
    with Invalid_argument _ -> false)
 
+let merlin_noloc = { txt = "merlin.loc"; loc = Location.none }, PStr []
+
+(* let to_js_of_ocaml = lazy (Some "Js_of_ocaml") *)
+let to_js_of_ocaml = lazy None
```
Comment (Drup) [inline lib/ppx/ppx_js.cppo.ml]: What is this for ? Packing jsoo like you do in janestreet ? (Could you add a comment ?)

Comment (Drup) [inline lib/ppx/ppx_js.cppo.ml]: Actually, we could directly add a ppx option, and pass the name of the module path prefix on the cli. I guess that would avoid issues for you.

Comment (hhugo) [inline lib/ppx/ppx_js.cppo.ml]: It's indeed used inside Janestreet. I'll remove this diff as it does not belong in the PR.

**Comment by Drup** [comment]
Can you give a detailed explanation of what `invoker` is supposed to be doing ? 

If I understand things correctly, we now use locally abstract types (with functions) instead of modules, and `invoker` generates the general code pattern that is used for all type constraints. The unifying aspect is good, but it means `invoker` should really be explained.

Also, could you give a quick idea of how/when ghost location and `{@merlin.loc]` are used ?

**Pull request merged**